lib/index.js

```diff
@@ -650,5 +665,21 @@ internals.findHeader = function (headerName, headers) {
     return foundKey && headers[foundKey];
 };
 
+internals.applyUrlToOptions = (options, url) => {
```

Comment:
Any thoughts on copying over the node version, which will also have the auth bits: https://github.com/nodejs/node/blob/master/lib/internal/url.js#L1257-L1276

Reply:
Yeah I initially did just pull in the Node function, then I pared it down to what the current implementation was as far as I could tell.

I'm all for including the auth bits if you'd like. I can add some tests around it, too. Was just trying to keep the initial pass at this as small as possible. :)